### PR TITLE
Add `gluonts.util.safe_extract`

### PR DESCRIPTION
**Pull request by lostella**
*Description of changes:* Similar to #2484, add `safe_extract` utility to sanitize destination paths before extracting archives.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Comment by lostella** [comment]
Ah! This can be simplified a lot

**Comment by jaheba** [comment]
Can we put this into `util`?

File: src/gluonts/util.py [hunk 1]
```diff
@@ -66,3 +67,29 @@ def my_property(self):
...
+    """
+    Safe wrapper around ``TarFile.extractall`` that checks all destination
+    files to be strictly within the given ``path``.
+    """
+    for member in tar.getmembers():
+        member_path = path / member.name
+        if not is_within_directory(path, member_path):
+            raise Exception("Attempted Path Traversal in Tar File")
+
+    tar.extractall(path, members, numeric_owner=numeric_owner)
```
Comment (jaheba) [inline src/gluonts/util.py]:
```suggestion
def safe_extract(
    tar, path: Path = Path("."), members=None, *, numeric_owner=False
):
    """
    Safe wrapper around ``TarFile.extractall`` that checks all destination
    files to be strictly within the given ``path``.
    """
    path = Path(path).resolve()

    for member in tar.getmembers():
        member_path = (path / member.name).resolve()

        try:
            member_path.relative_to(path)
        except ValueError:
            raise PermissionError(f"'{member.name}' extracts out of target.")

    tar.extractall(path, members, numeric_owner=numeric_owner)
```

**Review by jaheba — DISMISSED** [review]

**Review by jaheba — APPROVED** [review]

File: src/gluonts/nursery/tsbench/src/cli/evaluations/download.py [hunk 2]
```diff
@@ -19,6 +19,7 @@
 from typing import Any, cast, Dict, List, Optional
 import botocore
 import click
+from gluonts.util import safe_extract
```
Comment (jaheba) [inline src/gluonts/nursery/tsbench/src/cli/evaluations/download.py:22]: This need to be fixed.

**Pull request merged**